### PR TITLE
Issue #1 feat: docker compose files for keycloak

### PR DESCRIPTION
**Pull request by kartishr**

### CODE REVIEW

**Comment by devops-ekstep** [comment]
@kochhar please review

**Review by kochhar — CHANGES_REQUESTED** [review]
@devops-ekstep I have noted comments for changes which need to be made. Please address these comments and resubmit the PR.

File: keycloak/themes/sunbird/login/keycloak/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      POSTGRES_ROOT_PASSWORD: root_password
+    networks:
+        - default
+          
+  keycloak-1:
+    image: keycloak_image:3.2.1.Final
+    hostname: "keycloak-1"
+    environment:
+      KEYCLOAK_LOGLEVEL: DEBUG
+      POSTGRES_PORT_5432_TCP_ADDR: keycloak_db
```
Comment (kochhar) [inline keycloak/themes/sunbird/login/keycloak/docker-compose.yml]: Postgres data should be encrypted and come from a secrets file.

File: keycloak/themes/sunbird/login/keycloak/docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+KEYCLOAK_USER=admin
+KEYCLOAK_PASSWORD=ash12345678
```
Comment (kochhar) [inline keycloak/themes/sunbird/login/keycloak/docker-entrypoint.sh]: Keycloak credentials should not be part of this script or this commit. It is a security hole.

**Review by kochhar — CHANGES_REQUESTED** [review]

File: keycloak/deploy/docker/README [hunk 3]
```diff
@@ -0,0 +1,15 @@
+cd sunbird-auth/keycloak/keycloak/
+
+docker build -t "keycloak_image:3.2.1.Final" .
```
Comment (kochhar) [inline keycloak/deploy/docker/README]: @devops-ekstep aren't we currently using keycloak version 3.2.0.Final?

File: keycloak/deploy/docker/docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+
+KEYCLOAK_USER=admin
+KEYCLOAK_PASSWORD=ash12345^&*()
```
Comment (kochhar) [inline keycloak/deploy/docker/docker-entrypoint.sh]: passwords must be encrypted and vaulted.

File: keycloak/deploy/docker/docker-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+if [ $KEYCLOAK_USER ] && [ $KEYCLOAK_PASSWORD ]; then
+    keycloak/bin/add-user-keycloak.sh --user $KEYCLOAK_USER --password $KEYCLOAK_PASSWORD
+fi
+
+
+export HOSTNAME_IP=$(hostname -i)
+export HOSTNAME_IP_ALL=$(hostname --all-ip-addresses)
+echo "hostname -i returned: $HOSTNAME_IP, -I returned: $HOSTNAME_IP_ALL"
+
+exec /opt/jboss/keycloak/bin/standalone.sh -Djboss.bind.address.private=$HOSTNAME_IP $@
```
Comment (kochhar) [inline keycloak/deploy/docker/docker-entrypoint.sh]: this seems to be conflicting syntax with the CMD provided in the dockerfile. 

I suggest that 
1. you move the computation of the $HOSTNAME_IP to the Dockerfile and 
2. Add -bprivate $HOSTNAME_IP as additional args in the CMD line.

File: keycloak/deploy/docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,9 @@
+FROM jboss/keycloak-ha-postgres
+
+ADD docker-entrypoint.sh /opt/jboss/
```
Comment (kochhar) [inline keycloak/deploy/docker/Dockerfile]: Also need to change the standalone-ha.xml during the build to set
1. database configuration
1. logging configuration 
1. jgroups cluster configuration.

File: keycloak/deploy/docker/docker-entrypoint.sh [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+
+KEYCLOAK_USER=admin
+KEYCLOAK_PASSWORD=ash12345^&*()
+
+if [ $KEYCLOAK_USER ] && [ $KEYCLOAK_PASSWORD ]; then
```
Comment (kochhar) [inline keycloak/deploy/docker/docker-entrypoint.sh]: This must only be executed when a new environment is being created for the first time. This cannot be executed every time.

**Review by kochhar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**